src/mail/view/MailView.ts

```diff
@@ -450,8 +450,10 @@ export class MailView implements CurrentView {
 				}
 
 				const filteredFolders = folders.filter(f => f.mails !== mailList)
-				const targetFolders = getSortedSystemFolders(filteredFolders).concat(getSortedCustomFolders(filteredFolders))
-				return targetFolders.map(f => ({
+				return getSortedSystemFolders(filteredFolders)
+					.concat(getSortedCustomFolders(filteredFolders))
+					.filter(folder => allMailsCanBeInsideFolder(selectedMails, folder))
+					.map(f => ({
```

Comment:
The arrow function body needs to be indented, the formatting made my brain do a 720

Reply:
Oops, fixed. I have no idea how I missed that. Thank goodness for ctrl-s in WebStorm.